specification/dimensions/chargefrequency.md

```diff
@@ -0,0 +1,38 @@
+# Charge Frequency
+
+Charge Frequency indicates how often a charge will occur. Along with the Charge Period column, the Charge Frequency can be used to understand recurrence periods (e.g., monthly, yearly), forecast upcoming charges, and differentiate between one-time and recurring fees for purchases.
+
+The ChargeFrequency column MUST be present in the billing data and MUST NOT be null or empty. This column is of type String and MUST be one of the allowed values.
+
+## Column ID
+
+ChargeFrequency
+
+## Display Name
+
+Charge Frequency
+
+## Description
+
+Indicates how often a charge will occur.
+
+## Content Constraints
+
+| Constraint      | Value                                    |
+| :-------------- | :--------------------------------------- |
+| Column required | True                                     |
+| Data type       | String                                   |
+| Allows nulls    | False                                    |
+| Value format    | list-of-values                           |
+
+Allowed values:
+
+| Value       | Description                                                                                                                                                                                                                                      |
+| ----------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| Recurring   | This charge type will occur on a periodic cadence which can be daily, hourly or monthly which indicate charges for the acquisition of a service or resource bought upfront irrespective of their utilization on a recurring basis.               |
+| Usage-Based | Charges based on the quantity and frequency of a service or resource that was consumed in a given period of time. If the consumption/utilization of the service or resource stops, these charges will no longer be incurred.                     |
+| One-Time    | Charges that will occur only once and indicate a one-time purchase of a contract/commitment or pre-paid service. In other words, it can either be a pre-payment of usage charges or a truly one-time fee - such as a setup fee or a service fee. |                                       
```

Comment:
Reworded this for consistency and to focus on when the charge happens with expectations around the cadence. I removed the explicit examples since we have some in supporting content and they didn't cover all possible charges. I moved one-time first thinking that might flow more smoothly (once → multiple times → when used).
```suggestion
| Value       | Description                                                                                                                                                                                                                                                                         |
| ----------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| One-Time    | Charges that only happen once and will not repeat. One-time charges are typically recorded on the hour or day when the cost was incurred.                                                                                                                                           |
| Recurring   | Charges that repeat on a periodic cadence (e.g., weekly, monthly) regardless of whether the product or service was used. Recurring charges typically happen on the same day or point within every period. The charge date does not change based on how or when the service is used. |
| Usage-Based | Charges that repeat every time the service is used. Usage-based charges are typically recorded hourly or daily, based on the granularity of the cost data, but only when the service is used. Usage-based charges are not recorded when the service is not used.                    |
```